### PR TITLE
[shaderwriter] Update to version 2.0.0

### PR DESCRIPTION
**Pull request by DragonJoker**
Describe the pull request

    What does your PR fix?
    Updates shaderwriter to version 2.0.0

    Which triplets are supported/not supported? Have you updated the CI baseline?
    No change related to that, No

    Does your PR follow the maintainer guide?
    Hopefully yes

    If you have added/updated a port: Have you run ./vcpkg x-add-version --all and committed the result?
    Yes.

If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@DragonJoker ，Thanks for your pr, Could you please Modernize portfile.cmake? Use the latest cmake functions

**Comment by DragonJoker** [comment]
Where can I find the latest cmake functions list ?
Is it about vcpkg functions ? (can I rely on what's on this page : https://vcpkg.readthedocs.io/en/latest/maintainers/maintainer-guide/#portfiles ?)
Or about cmake itself ? (in that case, I'm not sure what could be upgraded ?)
Thanks

**Comment by JonLiu1993** [comment]
@DragonJoker , Please ping me if you have problems

**Comment by dg0yt** [comment]
> Where can I find the latest cmake functions list ?

The latest list is here in this repository. The offical web page is https://vcpkg.io/en/docs/maintainers/portfile-functions.html

**Comment by DragonJoker** [comment]
Is there something I can do about the failures here ?

File: ports/shaderwriter/portfile.cmake [hunk 1]
```diff
@@ -1,15 +1,17 @@
+set(VERSION 2.0.0)
 
 vcpkg_from_github(OUT_SOURCE_PATH SOURCE_PATH
     REPO DragonJoker/ShaderWriter
-    REF v1.1.0
-    SHA512 7c2a311f332342a2fc7d37e646314fe79f28a00292ff082903b3fc30f3ff86f222936fd2bc9d85d79a66caf854b05bfa5546f9d299c0764d695c027c5101bc74
+    REF c6bb7ee45ac378bb90d71ddb8d02488bf526be32
+    HEAD_REF master
+    SHA512 5aa5a4a001c3e5c00074c9706ccc7cbfa6ba010c71ba80d50b18115f84f584cdfdf8c735b19e2a7c727466cad143b70d56cff370b9a3c66118fd66c2d8e02b4e
```
Comment (JonLiu1993) [inline ports/shaderwriter/portfile.cmake:7]:
```suggestion
    SHA512 5aa5a4a001c3e5c00074c9706ccc7cbfa6ba010c71ba80d50b18115f84f584cdfdf8c735b19e2a7c727466cad143b70d56cff370b9a3c66118fd66c2d8e02b4e
    HEAD_REF master
```

Comment (JonLiu1993) [inline ports/shaderwriter/portfile.cmake:7]:
```suggestion
    SHA512 5aa5a4a001c3e5c00074c9706ccc7cbfa6ba010c71ba80d50b18115f84f584cdfdf8c735b19e2a7c727466cad143b70d56cff370b9a3c66118fd66c2d8e02b4e
    HEAD_REF master
```

**Review by JonLiu1993 — APPROVED** [review]

File: ports/shaderwriter/portfile.cmake [hunk 2]
```diff
@@ -1,15 +1,17 @@
+set(VERSION 2.0.0)
```
Comment (ras0219-msft) [inline ports/shaderwriter/portfile.cmake:1]: Where is this used?

Comment (JonLiu1993) [inline ports/shaderwriter/portfile.cmake:1]:
```suggestion
```

Comment (JonLiu1993) [reply ports/shaderwriter/portfile.cmake:1]: @DragonJoker ,Looks like this is not used, did you forget to add something? If not please delete it

Comment (DragonJoker) [reply ports/shaderwriter/portfile.cmake:1]: Ah, I'll remove it

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR and continuing to maintain a neat library!

**Pull request merged**